04_design_patterns/src/shipment.ts

```diff
@@ -0,0 +1,85 @@
+import { addMarks } from './add-marks';
+import { Marks, State } from './client';
+import { Shipper } from './shipper';
+import { AirEastShipper } from './shipper-strategies/air-east-shipper';
+import { ChigagoSprintShipper } from './shipper-strategies/chigago-sprint-shipper';
+import { PacificParcelShipper } from './shipper-strategies/pacific-parcel-shipper';
+import { Letter } from './shipper-visitors/letter';
+import { Oversized } from './shipper-visitors/oversized';
+import { Package } from './shipper-visitors/package';
+import { Visitor } from './shipper-visitors/visitor';
+
+export class Shipment {
+  private static shipmentAmount = 0;
+  private state: State;
+  private shipper!: Shipper;
+  private visitor!: Visitor;
+
+  constructor(state: State) {
+    this.state = state;
+
+    this.setStrategy(state.fromZipCode);
+    this.initializeVisitor(state.weight);
+  }
+
+  private setStrategy(fromZipCode: string): void {
+    if (['7', '8', '9'].includes(fromZipCode[0])) {
```

Comment:
@romasaldan , the task describes: "If the from zip code begins with 1, 2, or 3, then the source location is in the east, and Air East will be used"

There should be `.startsWith`